packages/devextreme-cli/templates/react/application/src/app-routes.js

```diff
@@ -14,3 +15,9 @@
     component: HomePage
   }
   <%=/empty%>];
+
+routes.forEach(route => {
+  route.component = withNavigationWatcher(route.component);
```

Comment:
I suggest moving withNavigationWatcher here, as it is used once. I find this more clear:
```js
routes.forEach(route => {
  route.component = function (props) {
    const { path } = props.match;
    const { setNavigationData } = useContext(NavigationContext);

    useEffect(() => {
      setNavigationData({ currentPath: path });
    }, [path, setNavigationData]);

    return React.createElement(route.component, props);
  }
});
```